pkg/list/list_stacks.go

```diff
@@ -1,45 +1,365 @@
+		// Add custom template functions
+		funcMap := template.FuncMap{
+			"getVar": func(vars map[string]any, key string) string {
+				if val, ok := vars[key]; ok && val != nil {
+					return fmt.Sprintf("%v", val)
+				}
+				return ""
+			},
+		}
+
+		tmpl, err := template.New(col.Name).Funcs(funcMap).Parse(col.Value)
+		if err != nil {
+			return "", fmt.Errorf("error parsing template for column %s: %w", col.Name, err)
+		}
+		templates[i] = columnTemplate{name: col.Name, template: tmpl}
 	}
 
-	// List all stacks
-	stacks := lo.Keys(stacksMap)
-	sort.Strings(stacks)
-	return strings.Join(stacks, "\n") + "\n", nil
+	// Process each stack and populate rows
+	for i, stack := range filteredStacks {
+		row := make([]string, len(listConfig.Columns))
+		for j, tmpl := range templates {
+			var buf strings.Builder
+			if err := tmpl.template.Execute(&buf, stack); err != nil {
+				return "", fmt.Errorf("error executing template for column %s: %w", tmpl.name, err)
+			}
+			// Just use the raw string value
+			row[j] = buf.String()
+		}
+		rows[i] = row
+	}
+
+	// Handle different output formats
+	switch format {
+	case FormatJSON:
+		// Convert to JSON format using only non-empty columns
+		var stacks []map[string]string
+		for _, row := range rows {
+			s := make(map[string]string)
+			for i, header := range headers {
+				// Only include non-empty values in JSON output
+				if row[i] != "" {
+					// Use raw value without any color formatting for JSON output
+					s[strings.ToLower(header)] = row[i]
+				}
+			}
+			stacks = append(stacks, s)
+		}
+		// Use plain JSON marshaling without any color formatting
+		jsonBytes, err := json.MarshalIndent(stacks, "", "  ")
+		if err != nil {
+			return "", fmt.Errorf("error formatting JSON output: %w", err)
+		}
+		return string(jsonBytes) + utils.GetLineEnding(), nil
+
+	case FormatCSV, FormatTSV:
+		// Only include columns that have values
+		var nonEmptyHeaders []string
+		var nonEmptyColumnIndexes []int
+
+		// Find columns that have at least one non-empty value
+		for i, header := range headers {
+			hasValue := false
+			for _, row := range rows {
+				if row[i] != "" {
+					hasValue = true
+					break
+				}
+			}
+			if hasValue {
+				nonEmptyHeaders = append(nonEmptyHeaders, header)
+				nonEmptyColumnIndexes = append(nonEmptyColumnIndexes, i)
+			}
+		}
+
+		// Set appropriate delimiter based on format
+		fileDelimiter := delimiter
+		if delimiter == "\t" || delimiter == "" {
+			switch format {
+			case FormatCSV:
+				fileDelimiter = ","
+			case FormatTSV:
+				fileDelimiter = "\t"
+			}
+		}
+
+		var output strings.Builder
+
+		// Helper function to escape values
+		escapeValue := func(s string) string {
+			// For TSV, just replace tabs with spaces to maintain format
+			if format == FormatTSV {
+				return strings.ReplaceAll(s, "\t", " ")
+			}
+			// For CSV, use standard CSV escaping
+			if strings.Contains(s, fileDelimiter) || strings.Contains(s, "\n") || strings.Contains(s, "\"") {
```

Comment:
This is super hacky. We should be using the idiomatic way of creating CSV formatted output using ``encoding/csv`